url7f729337e3_timmolter_Sundial_tgz-pJ8-org_knowm_sundial_SampleRunJ8/src/org/knowm/sundial/plugins/AnnotationJobTriggerPlugin.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright 2015 Knowm Inc. (http://knowm.org) and contributors.
+ * Copyright 2013-2015 Xeiam LLC (http://xeiam.com) and contributors.
+ * Copyright 2001-2011 Terracotta Inc. (http://terracotta.org).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.knowm.sundial.plugins;
+
+import static org.quartz.builders.CronTriggerBuilder.cronTriggerBuilder;
+import static org.quartz.builders.JobBuilder.newJobBuilder;
+import static org.quartz.builders.SimpleTriggerBuilder.simpleTriggerBuilder;
+
+import java.text.ParseException;
+import java.util.Set;
+import java.util.TimeZone;
+
+import org.knowm.sundial.Job;
+import org.knowm.sundial.annotations.CronTrigger;
+import org.knowm.sundial.annotations.SimpleTrigger;
+import org.quartz.core.Scheduler;
+import org.quartz.exceptions.SchedulerException;
+import org.quartz.jobs.JobDataMap;
+import org.quartz.jobs.JobDetail;
+import org.quartz.plugins.SchedulerPlugin;
+import org.quartz.triggers.OperableTrigger;
+import org.quartz.triggers.Trigger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This plugin adds jobs and schedules them with triggers from annotated Job classes as the scheduler is initialized.
+ *
+ * @author timmolter
+ */
+public class AnnotationJobTriggerPlugin implements SchedulerPlugin {
+
+  private final Logger logger = LoggerFactory.getLogger(AnnotationJobTriggerPlugin.class);
+
+  private Scheduler scheduler;
+
+  private final String packageName;
+
+  private static final String SEPARATOR = ":";
+
+  /**
+   * Constructor
+   *
+   * @param packageName
+   */
+  public AnnotationJobTriggerPlugin(String packageName) {
+
+    this.packageName = packageName;
+  }
+
+  /**
+   * <p>
+   * Called during creation of the <code>Scheduler</code> in order to give the <code>SchedulerPlugin</code> a chance to initialize.
+   * </p>
+   *
+   * @throws org.quartz.exceptions.SchedulerConfigException if there is an error initializing.
+   */
+  @Override
+  public void initialize(String name, final Scheduler scheduler) throws SchedulerException {
+
+    logger.info("Initializing AnnotationJobTriggerPlugin Plug-in.");
+
+    this.scheduler = scheduler;
+  }
+
+  @Override
+  public void start() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.